test/e2e/test/helper/yaml.go

```diff
@@ -271,7 +276,7 @@ func transformToE2E(namespace, fullTestName, suffix string, transformers []Build
 				WithKibanaRef(tweakServiceRef(b.Beat.Spec.KibanaRef, suffix))
 
 			if b.PodTemplate.Spec.ServiceAccountName != "" {
-				b = b.WithPodTemplateServiceAccount(b.PodTemplate.Spec.ServiceAccountName + "-" + suffix)
+				b = b.WithPodTemplateServiceAccount(b.PodTemplate.Spec.ServiceAccountName + "-" + suffix) //nolint:wastedassign
```

Comment:
It works because `b.PodTemplate` is a pointer because it's a bit tricky. Don't yo`builder = builder.WithPodTemplateServiceAccount`

Reply:
Good point!